packages/components/src/components/validate/validate.component.ts

```diff
@@ -0,0 +1,268 @@
+import type { CSSResultGroup } from 'lit';
+import { html } from 'lit';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import componentStyles from '../../styles/component.styles.js';
+import SynergyElement from '../../internal/synergy-element.js';
+import { watch } from '../../internal/watch.js';
+import SynAlert from '../alert/alert.component.js';
+import {
+  arraysDiffer,
+  isBlurEvent,
+  isInvalidEvent,
+} from './utility.js';
+import styles from './validate.styles.js';
+
+/**
+ * @summary Validate is a helper that may be used to wrap
+ * synergy input fields and forms to provide validation message.
+ *
+ * @dependency syn-alert
+ *
+ * @slot - The input fields or form element to be validated.
+ * Avoid slotting in more than one element, as subsequent ones will be ignored.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart input-wrapper - The container that wraps the input field.
+ * @csspart alert - The container that wraps the alert.
+ * @csspart alert-message - The container that wraps the alert message.
+ */
+export default class SynValidate extends SynergyElement {
+  static styles: CSSResultGroup = [componentStyles, styles];
+
+  static dependencies = {
+    'syn-alert': SynAlert,
+  };
+
+  controller = new AbortController();
+
+  @queryAssignedElements() private slottedChildren: HTMLElement[];
+
+  @state() validationMessage = '';
+
+  /** Show the validation message underneath the element, using a syn-alert */
+  @property({ reflect: true, type: Boolean }) inline = false;
+
+  /** Do not show the error icon when using inline validation */
+  @property({ attribute: 'hide-icon', reflect: true, type: Boolean }) hideIcon = false;
+
+  /**
+   * Defines the events that trigger the validation.
+   * Defaults to the `invalid` and `change` events.
+   * `invalid` will always automatically be included.
+   * You may also use the `live` keyword to validate on every input change.
+   * @example ```html
+   * <!-- Use default validation (invalid, change) -->
+   * <syn-validate></syn-validate>
+   *
+   * <!-- Validate on invalid and change events (invalid, change) -->
+   * <syn-validate on="invalid change"></syn-validate>
+   *
+   * <!-- Validate on live events (invalid, blur, input)-->
+   * <syn-validate on="live"></syn-validate>
+   *
+   * <!-- Validate on live and custom events (invalid, focus, input, change) -->
+   * <syn-validate on="live focus change"></syn-validate>
+   * ```
+   */
+  @property({
+    converter: {
+      fromAttribute: (e: string) => e.split(' ').map(s => s.trim()),
+      toAttribute: (a: string[] | string) => {
+        if (!Array.isArray(a)) return a;
```

Comment:
If the input given to `on` as property is a string, it should not just be returned but also the same should be done as in the fromAttribute `a.split(' ').map(s => s.trim())` before returning it.

Because as a user, when I see the description of the `on` property I would assume I can either add a string oder a string array value.

![image](https://github.com/user-attachments/assets/936cb0fd-66f4-4c96-9649-2e72f826b75b)

Everything works fine if adding only a single event as string:

```js
      const validate = document.querySelector('syn-validate');
      setTimeout(() => {
        validate.on = 'change';
      }, 3000);
```

The events array will then look like following which is correct: ` ['syn-change', 'syn-invalid']`

But if adding more than one event as a string like following:

```js
      const validate = document.querySelector('syn-validate');
      setTimeout(() => {
        validate.on = 'change blur';
      }, 3000);
```

The events array will then look like following which is incorrect: ` ['syn-change blur', 'syn-invalid']`

Reply:
After thinking deeper about it, I now use a string always. All frameworks have custom syntax making it unneccesairy hard to use this. Also, we can get rid of most of the logic when sticking to a simple string.